libV2/cache/backend/redis.js

```diff
@@ -176,6 +177,44 @@ class RedisCache {
             throw error;
         }
     }
+
+    async updateCounters(metric) {
+        if (metric.sizeDelta) {
+            try {
+                const accountSizeKey = schema.getAccountSizeCounterKey(this._prefix, metric.account);
+                await this._redis.incrby(accountSizeKey, metric.sizeDelta);
+            } catch (error) {
+                moduleLogger
+                    .with({ method: 'updateCounter' })
+                    .error('error while updating metric counters', { error });
+                throw error;
+            }
+        }
+    }
+
+    async updateAccountCounterBase(account, size) {
+        try {
+            const accountSizeKey = schema.getAccountSizeCounterKey(this._prefix, account);
+            const accountSizeBaseKey = schema.getAccountSizeCounterBaseKey(this._prefix, account);
+            await this._redis.mset(accountSizeKey, 0, accountSizeBaseKey, size);
+            await this._redis.expire(accountSizeBaseKey, constants.counterBaseValueExpiration);
+        } catch (error) {
+            moduleLogger
+                .with({ method: 'updateAccountCounterBase' })
+                .error('error while updating metric counter base', { error });
+            throw error;
+        }
```

Comment:
I have seen this or something very similar a bit and I think its an alternative method would be to use a decorator. Although javascript decorators are still a proposal you could wrap it yourself easily enough. Consider if say the module logger had a method like this:

```javascript
function errorLogger(fn, description) {
    return function(...params) {
        try {
            fn(...params);
        } catch (error) {
            this.with({method: fn.name })
            this.error(description, { error } );
        }
    }
}
```
( would need one for async that returns an async function but you get the point.

Then your function could just be:

```javascript
updateAccountCounterBase = moduleLogger.asyncLogger((account, size) => {
    const accountSizeKey = schema.getAccountSizeCounterKey(this._prefix, account);
    const accountSizeBaseKey = schema.getAccountSizeCounterBaseKey(this._prefix, account);
    await this._redis.mset(accountSizeKey, 0, accountSizeBaseKey, size);
    await this._redis.expire(accountSizeBaseKey, 
}, 'error while updating metric counter base')
```
Note that after the decorator proposal you would write
```javascript
@moduleLogger('error while updating metric counter base')
async function(...)
```

If you have used python decorators it seems JS decorators are the same. For reference here is a typescript guide about them https://www.typescriptlang.org/docs/handbook/decorators.html.

Reply:
I like the idea of removing the try/catch bolierplate, I'll see what I can do

Reply:
After digging into it, public class fields (this syntax `updateAccountCounterBase = moduleLo....`) is also an open proposal so using decorators would require something like
```javascript
class TestClass {
    constructor() {
        this._data = { val: 'world' };
    }

    _myFunc() {
        console.log(`hello ${this._data.val}`);
    }
}

TestClass.prototype.testmeOut = function () { this._myFunc(); };
```
which is less than ideal, or by moving to a more traditional way of building js classes which would be in contrast to the rest of the project. I instead went with a more traditional wrapper. `getKey/setKey` have been updated to use it, let me know what you think.

Reply:
Hmm... dang JS step your game up. Let me see.

Reply:
Well your implementation seems like a good alternative but doesn't seem dramatically better than try/catch so I am not sure it is worth going against all the other parts that use try/catch. Maybe if decorators are approved we can make some sweet wrappers ( I love decorators from python ). Looks good to me either way though.